### PR TITLE
Refactor Jetpack Options

### LINKED ISSUE #7160: Jetpack_Sync_Options should be wrapped in Jetpack_Options
Based on a slack conversation with @lezama - we should move the methods in `Jetpack_Sync_Options` to `Jetpack_Options` and rename to them to, for example, `get_raw_option`. This will make them clearly re-usable.

### PR DESCRIPTION
**Pull request by roccotripaldi**
We can remove class.jetpack-sync-options.php and move it's methods to Jetpack_Options class. 

Getting / Updating / Deleting a raw option is not a concept that is only required by sync, so it was confusing to have a sync prefix.

Fixes #7160

### CODE REVIEW

**Comment by georgestephanis** [comment]
Just a reminder that `class.jetpack-options.php` is synced to the wpcom codebase and will need to be merged there after this goes to master.

File: class.jetpack-options.php [hunk 1]
```diff
@@ -321,4 +320,59 @@ private static function delete_grouped_option( $group, $names ) {
 		return true;
 	}
 
+	static function delete_raw_option( $name ) {
+		global $wpdb;
+		$wpdb->query( $wpdb->prepare( "DELETE FROM $wpdb->options WHERE option_name = %s", $name ) );
+	}
+
+	// Raw option methods allow Jetpack to get / update / delete options via direct DB queries, including options
```
Comment (georgestephanis) [inline class.jetpack-options.php]: Should this comment be above the `delete_raw_option()` method?

File: class.jetpack-options.php [hunk 2]
```diff
@@ -321,4 +320,59 @@ private static function delete_grouped_option( $group, $names ) {
 		return true;
 	}
 
+	static function delete_raw_option( $name ) {
```
Comment (georgestephanis) [inline class.jetpack-options.php]: function needs phpdoc

File: class.jetpack-options.php [hunk 3]
```diff
@@ -321,4 +320,59 @@ private static function delete_grouped_option( $group, $names ) {
...
+		global $wpdb;
+		$wpdb->query( $wpdb->prepare( "DELETE FROM $wpdb->options WHERE option_name = %s", $name ) );
+	}
+
+	// Raw option methods allow Jetpack to get / update / delete options via direct DB queries, including options
+	// that are not created by the Jetpack plugin. This is helpful when we need to bypass cache and filters.
+
+	static function update_raw_option( $name, $value, $autoload = false ) {
```
Comment (georgestephanis) [inline class.jetpack-options.php]: function needs phpdoc

File: class.jetpack-options.php [hunk 4]
```diff
@@ -321,4 +320,59 @@ private static function delete_grouped_option( $group, $names ) {
...
+
+		$autoload_value = $autoload ? 'yes' : 'no';
+
+		// we write our own option updating code to bypass filters/caching/etc on set_option/get_option
+		global $wpdb;
```
Comment (georgestephanis) [inline class.jetpack-options.php]: I normally prefer `global` to be the first line in a function so it doesn't get missed, but I'm not sure if its a codified standard.  Probably fine, but worth mentioning.

File: class.jetpack-options.php [hunk 5]
```diff
@@ -321,4 +320,59 @@ private static function delete_grouped_option( $group, $names ) {
...
+					"INSERT IGNORE INTO $wpdb->options ( option_name, option_value, autoload ) VALUES ( %s, %s, '$autoload_value' )",
+					$name,
+					$serialized_value
+				)
+			);
+		}
+		return $updated_num;
+	}
+
+	static function get_raw_option( $name, $default = null ) {
```
Comment (georgestephanis) [inline class.jetpack-options.php]: needs phpdoc

**Comment by roccotripaldi** [comment]
> Does the new jetpack_secrets thing needs a routine to clean old secret related options?

Nice catch. Addressed in 4f0bf5c7fe6851b7b4bde2a779a5e0a97029471c

**Pull request merged**